memcached-operator/pkg/controller/memcached/memcached_controller.go

```diff
@@ -122,15 +130,30 @@ func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Res
 
 	// Ensure the deployment size is the same as the spec
 	size := memcached.Spec.Size
-	if *found.Spec.Replicas != size {
-		found.Spec.Replicas = &size
-		err = r.client.Update(context.TODO(), found)
+	if *deployment.Spec.Replicas != size {
+		deployment.Spec.Replicas = &size
+		err = r.client.Update(context.TODO(), deployment)
 		if err != nil {
-			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", found.Namespace, "Deployment.Name", found.Name)
+			reqLogger.Error(err, "Failed to update Deployment.", "Deployment.Namespace", deployment.Namespace, "Deployment.Name", deployment.Name)
 			return reconcile.Result{}, err
 		}
-		// Spec updated - return and requeue
-		return reconcile.Result{Requeue: true}, nil
+	}
+
+	// Check if the service already exists, if not create a new one
```

Comment:
What is the point of this Service? Can we add a comment explaining this to the users.

Reply:
Make more clear how things work only by providing the creating of more one common object. Also, note the with it we could make clear the usage of `res, err = r.Reconcile(req)` in the test as well. We can add some info as follows, please let me know wdyt?

```suggestion
         // Check if the service already exists, if not create a new one
         //  NOTE: The Service is used to expose the deployment. However, the service is not required at all for this solution. It was just added with the purpose to bring more clarity over how you can do your own operators-projects.   
```

Reply:
Done 👍